compiler+runtime/src/cpp/jank/read/lex.cpp

```diff
@@ -289,7 +336,15 @@ namespace jank::read
       }
 
       auto const token_start(pos);
-      switch(file[token_start])
+      auto const oc = convert_to_codepoint(file.substr(token_start), token_start);
+      char32_t start{};
+      size_t len{};
+      if (oc.is_ok())
+      {
+        start = oc.expect_ok().character;
+        len = oc.expect_ok().len;
+      }
```

Comment:
What happens if it's not ok? We're checking the wrong thing here. There's a pattern in jank's code which you may not have noticed yet.

Basically every `if` check in jank's code is checking for the negative case. This is for two reasons:

1. It documents and asserts our invariants up front, allowing us to fail fast
2. It keeps the successful path *less nested*

For the second point, nesting code is considered a bad smell. If you find yourself checking for a positive if condition, pause, since you should probably swap it around.

In this case, the whole reason we returned an error from `convert_to_codepoint` was so that we can return it as an error here. With the code as it is, that error gets swallowed. There's a test we could write, which has an error with the starting char, which should then fail since that error isn't returned.